lib/config/fileconf.go

```diff
@@ -778,7 +778,8 @@ type StaticToken string
 // apart and constructs a services.ProvisionToken which contains the token,
 // role, and expiry (infinite).
 func (t StaticToken) Parse() (*types.ProvisionTokenV1, error) {
-	parts := strings.Split(string(t), ":")
+	// Split only on the first ':', for future cross platform compat with windows paths
```

Comment:
This is a nice change, but let's get a test for it as well.

Reply:
I'm going to test this by using a "real looking" windows path, but it won't actually attempt to read the token file since the testing is happening on a linux box. The important thing is we parse the path correctly.

Reply:
```go
func TestAuthenticationConfig_Parse_StaticToken(t *testing.T) {
	roles := []string{"Auth", "Node", "Proxy"}
	joinedRoles := strings.Join(roles[:], ",")
	var token StaticToken = StaticToken(fmt.Sprintf("%s:token", joinedRoles))

	provisionToken, err := token.Parse()
	require.NoError(t, err)

	require.Len(t, provisionToken.Roles, len(roles))
	for i := range roles {
		require.Equal(t, string(provisionToken.Roles[i]), roles[i])
	}
	require.Equal(t, provisionToken.Token, "token")

	// test parsing a windows filepath which contains a colon
	windowsTokenPath := "C:\\path\\to\\some\\token\\file"
	token = StaticToken(fmt.Sprintf("%s:%s", joinedRoles, windowsTokenPath))

	provisionToken, err = token.Parse()
	require.NoError(t, err)

	require.Len(t, provisionToken.Roles, len(roles))
	for i := range roles {
		require.Equal(t, string(provisionToken.Roles[i]), roles[i])
	}
	require.Equal(t, provisionToken.Token, windowsTokenPath)
}
```
Added test and verified it passed.